packages/sdk/src/index.ts

```diff
@@ -25,25 +26,18 @@ interface ScheduleXcmpTaskParams {
     overallWeight?: Weight;
     executionFeeAmount?: BN;
   };
+  scheduleAs?: HexString;
   keyringPair: KeyringPair;
 }
 
-interface ScheduleXcmpTaskWithPayThroughSoverignAccountFlowParams extends ScheduleXcmpTaskParams {
-  destinationChainAdapter: ChainAdapter;
-}
-
+/**
+ * The params for scheduleXcmpTaskWithPayThroughRemoteDerivativeAccountFlow
+ * invoker is the chain adapter that will send the XCM message to Turing/OAK to schedule the task
+ * invokeFeeLocation is the location of the fee asset for XCM message execution
+ */
 interface ScheduleXcmpTaskWithPayThroughRemoteDerivativeAccountFlowParams extends ScheduleXcmpTaskParams {
-  destinationChainAdapter: TaskSchedulerChainAdapter;
-  scheduleAs?: HexString;
-}
-
-interface CreateTaskFuncParams {
-  oakApi: ApiPromise;
-  destination: any;
-  executionFee: any;
-  encodedCall: HexString;
-  encodedCallWeight: Weight;
-  overallWeight: Weight;
+  invoker: TaskSchedulerChainAdapter;
+  invokeFeeLocation: any;
```

Comment:
let’s use `caller` instead of `invoker`, and `callerXcmFeeLocation` to replace `invokeFeeLocation`